File: packages/react-native-nitro-test-external/android/src/main/java/com/margelo/nitro/test/external/HybridSomeExternalObject.kt
package com.margelo.nitro.test.external

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridSomeExternalObject : HybridSomeExternalObjectSpec() {
  override fun getValue(): String {
    return "Hello world!"
  }

  override fun createOptionalPrimitivesHolder(
    optionalNumber: Double?,
    optionalBoolean: Boolean?,
    optionalUInt64: ULong?,
    optionalInt64: Long?,
  ): OptionalPrimitivesHolder {
    return OptionalPrimitivesHolder(optionalNumber, optionalBoolean, optionalUInt64, optionalInt64)
  }
}

File: packages/react-native-nitro-test-external/ios/HybridSomeExternalObject.swift
//
//  HybridSomeExternalObject
//  NitroTestExternal
//
//  Created by Marc Rousavy on 14.08.25.
//

import NitroModules

class HybridSomeExternalObject: HybridSomeExternalObjectSpec {
  func getValue() throws -> String {
    return "Hello world!"
  }

  func createOptionalPrimitivesHolder(
    optionalNumber: Double?, optionalBoolean: Bool?, optionalUInt64: UInt64?, optionalInt64: Int64?
  ) -> OptionalPrimitivesHolder {
    return OptionalPrimitivesHolder(
      optionalNumber: optionalNumber,
      optionalBoolean: optionalBoolean,
      optionalUInt64: optionalUInt64,
      optionalInt64: optionalInt64
    )
  }
}
